libs/wire-api-federation/proto/router.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+package wire.federator;
+
+// These definitions are part of the federation protocol.
+//
+// A component (e.g. 'brig') will send some data of type 'RemoteCall' to
+// the local 'federator' service. That federator will extract the domain from
+// the 'RemoteCall' to discover where to send some data of type 'LocalCall' to.
+// On the other end, the 'LocalCall' will be converted to a 'Response'.
+
+enum Component {
+  Brig = 0;
+}
+
+message HTTPResponse {
+    uint32 responseStatus = 1;
+    bytes responseBody = 2;
+}
+
+message Response {
+  oneof response {
+    HTTPResponse httpResponse = 1;
+    string err = 2;
+  }
+}
+
+// the envelope thing which is sent from brig to local federator
+message RemoteCall {
+  string domain = 1;
+  LocalCall localCall = 2;
+}
+
+// the federator will take out LocalCall from the RemoteCall Envelope, and
+// forward the LocalCall object to a federator on another backend (with the wish
+// that this remote federator forwards this to its own local brig)
+message LocalCall {
+  Component component = 1;
+  Method method = 2;
+  bytes path = 3;
+  repeated QueryParam query = 4;
+  bytes body = 5;
+}
+
+message QueryParam {
+  bytes key = 1;
+  bytes value = 2;
+}
+
+enum Method {
+  GET = 0;
+  POST = 1;
+  HEAD = 2;
+  PUT = 3;
+  DELETE = 4;
+  TRACE = 5;
+  CONNECT = 6;
+  OPTIONS = 7;
+  PATCH = 8;
+}
+
+// The * part of the following network traffic:
+// brig *-> federator -> federator -> brig
+// FUTUREWORK(federation): better naming?
+service RouteToRemote {
+  rpc call (RemoteCall) returns (Response);
+}
+
+// The * part of the following network traffic:
+// Brig -> federator *-> federator -> brig
+// FUTUREWORK(federation): better naming?
+service RouteToInternal {
```

Comment:
Could we simply call these services InternalService and ExternalService ? Or InternalFederator and ExternalFederator? This would match the port naming; and brig would talk to InternalFederator and give it a QualifiedRequest, and then the federator would send an UnqualifiedRequest to the ExternalFederator service of a federator of another backend.

Reply:
I like InternalFederator and ExternalFederator. I think it will still be confusing for somebody new to this because it is never clear if InternalFederator or InternalService means that it is routing **from** internal things or **to** internal things. But it will at least be consistent and there will be fewer confusing words in the domain.

Reply:
When this question arises, I usually find that the easiest to understand is to name things from the perspective of the code, or the service running the code.
 - I understand this is an interface, so technically it runs on both ends, but I think the diagram is clear that the Federator is the star of the show. 
 - Also I think it's best to keep with one set of terms either `Local/Remote` or `InternalExternal`, since `Local/Remote` is usually for "not across some network" or "across some network" and for microservices everything is "across some network", I would go for `Internal/External`. 

So something like `FromInternalService`/ `InternalServiceRoute` or `InternalServiceRPC` (my favorite) and `ToExternalFederator` or `RemoteFederatorRoute` or `RemoteFederatorRPC` (my favorite) would be easiest for me to understand.

Reply:
renamed as discussed.